### PR TITLE
Merge the various data maps and lists into a unified structure | test(torchlib)

### PR DESCRIPTION
**Pull request by justinchuby**
- Create `TorchLibOpInfo` for putting everything about an op in a single place.
- Fix `dtype_op_schema_compatible` to account for sequence types.
- Fix `_aten_var_mean_dim_onnx` to support float16
- I removed the allow list in favor of the xfail and skips. In a next PR we will auto create new xfails and skips when we enable tests for new data types

### CODE REVIEW

**Comment by justinchuby** [comment]
TODO: Fix

SUBSKIP [1] onnxscript/tests/function_libs/torch_lib/ops_test.py:361: dtype 'torch.float32' is not supported by the op 'atleast_2d'. Type constraints: [OpSchema.TypeConstraintParam(type_param_str='TTensor', allowed_type_strs=['tensor(bfloat16)', 'tensor(bool)', 'tensor(complex128)', 'tensor(complex64)', 'tensor(double)', 'tensor(float)', 'tensor(float16)', 'tensor(int16)', 'tensor(int32)', 'tensor(int64)', 'tensor(int8)', 'tensor(uint8)'], description=''), OpSchema.TypeConstraintParam(type_param_str='Sequence_TTensor', allowed_type_strs=['seq(tensor(bfloat16))', 'seq(tensor(bool))', 'seq(tensor(complex128))', 'seq(tensor(complex64))', 'seq(tensor(double))', 'seq(tensor(float))', 'seq(tensor(float16))', 'seq(tensor(int16))', 'seq(tensor(int32))', 'seq(tensor(int64))', 'seq(tensor(int8))', 'seq(tensor(uint8))'], description='')]
SUBSKIP [1] onnxscript/tests/function_libs/torch_lib/ops_test.py:361: dtype 'torch.float16' is not supported by the op 'log_softmax'. Type constraints: [OpSchema.TypeConstraintParam(type_param_str='T0', allowed_type_strs=['tensor(float)'], description='')]

**Comment by justinchuby** [comment]
TODO: allow eager mode to auto cast attributes (int->float)

```
E   RuntimeError: Unable to create onnxruntime InferenceSession for executing .RandomNormalLike op with onnx model
E   <
E      ir_version: 3,
E      opset_import: ["" : 1]
E   >
E   node_graph (float16[2,3] input0) => (float16[2,3] output0) {
E      output0 = RandomNormalLike <mean = 1.1, scale = 1> (input0)
E   }
```

**Comment by fatcat-z** [comment]
Looks like the CastLike has an issue in _aten_var_mean_dim_onnx function which caused CI failures.

**Review by titaiwangms — APPROVED** [review]
So right now, float16 and float32 are both supported in the test with skip and xfail?

**Comment by justinchuby** [comment]
That's right

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 1]
```diff
@@ -55,6 +61,100 @@
...
+@dataclasses.dataclass
+class TorchLibOpInfo:
+    """A dataclass to store the information to test an torchlib op."""
+
+    # The name of the op_info, e.g. "add"
+    op_info_name: str
+    # The torchlib ONNX Function to test
+    op: Callable[..., Any]
+    # Explicitly specify when the op is trace_only
+    trace_only: bool = False
```
Comment (titaiwangms) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:73]: Why do we want this explicitly?

**Pull request merged**